Misc fixes to OVS

cathay4t opened the pull request:
* Resolve unknown interface type to ovs bridge.
 * Fix error when modifying existing ovs patch interface.
 * Fix adding new ovs interface to existing bridge.

cathay4t committed:
ovs: Fix the interface type resolve code for ovs bridge

Then desire state does not define `type: ovs-bridge`, nmstate current
code treat it as new interface instead of searching for existing user
space interface.

The root cause is we incorrectly assume unknown interface as user space
interface.

The fix is remove old unknown interface from `Interfaces.kernel_ifaces`
and resolve its type using user space interfaces also.

Unit test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera commented on rust/src/lib/iface.rs:
```diff
@@ -113,7 +113,7 @@ impl InterfaceType {
         [Self::Bond, Self::LinuxBridge, Self::OvsBridge, Self::Vrf];
 
-    // Unknown and other interfaces are also considered as userspace
+    // other interfaces is also considered as userspace
```
small typo, "other interfaces **are** also considered.."

cathay4t committed:
nm ovs: Only generate NmSettingOvsIface when empty

When user try to modify an existing OVS patch port, nmstate will fail
with error:

    A connection with 'ovs-patch' setting needs to be of 'patch'
    interface type, not 'internal'"

The root cause is we override NmSettingOvsIface with default `internal`
ovs interface type. The fix is only generate NmSettingOvsIface when
empty or modified.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
ovs: Fix adding new ovs interface to existing bridge

When adding new ovs interface to existing ovs bridge, nmstate will fail
with error:
    Bug: BUG: Failed to find UUID of controller connection: ovs2, ovs-port

This is because we place this new ovs interface into `new_ifaces` while
its controller OVS bridge into `chg_ifaces`, this then lead to NM plugin
does not create new OVS port connection.

The fix is just place new OVS interface to changed interface list when
its controller OVS bridge is found in current state.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
The reason we passed CI before is old code always create new OVS bridge connection instead of loading existing one which hide this problems.
